database/src/postgres/mod.rs

```diff
@@ -38,6 +39,7 @@ pub struct PostgresDBManager {
     shards_pool:
         std::collections::HashMap<near_primitives::types::ShardId, sqlx::Pool<sqlx::Postgres>>,
     meta_db_pool: sqlx::Pool<sqlx::Postgres>,
+    earliest_available_block: Option<near_primitives::types::BlockHeight>,
```

Comment:
Could you please elaborate on the concept of this naming. And in a different place this the same name set to `None` means archival node mode. I am a bit lost.

Reply:
The main idea is to check block_height and avoid doing queries into the database.  Moved this into the configuration.